src/org/purefn/egon/api.clj

```diff
@@ -21,13 +21,20 @@
   [s3 bucket key value & {:keys [meta]}]
   (proto/store s3 bucket key value meta))
 
+(defn parse-opts [opts]
+  (if (even? (count opts))
+    (apply hash-map opts)
+    (set opts)))
+
 (defn fetch
-  "Fetches a document from an s3 bucket.  Accepts an optional sequence of keywords:
+   "Fetches a document from an s3 bucket.  Accepts optional args:
   :meta - indicates whether to return metadata long with the document content.
           When supplied, and map is returned with the the document as the :content
-          key and metadata as other keys in the map."
+          key and metadata as other keys in the map. Can be passed as single keyword if
+          no other args are used, otherwise follow by true or false.
+   :version-id - follow by S3 version-id to fetch a particular version"
   [s3 bucket key & opts]
-  (proto/fetch s3 bucket key (set opts)))
+  (proto/fetch s3 bucket key (parse-opts opts)))
```

Comment:
Passing a optional map would've been better, but bit of a hack instead to retain backward compatibility.